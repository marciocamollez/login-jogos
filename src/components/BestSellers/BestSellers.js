import React from "react";
import { Button } from "react-bootstrap";
import { Link } from "react-router-dom";
import styles from "./BestSellers.module.scss";

function BestSellers({ best }) {
  return (
    <div>
      <section className={styles.maisVendidos} data-testid="test-bestsellers">
        {best &&
          best.map((game) => (
            <div key={game.id} className={styles.container}>
              <div className={styles.boxImage}>
                <img src={game.img} alt={game.nome} title={game.nome} />
                <div className={styles.boxOverlay}>
                  <div className={styles.boxHide}>
                    <div>
                      <p>{game.nome}</p>
                      <p className={styles.preco}>{game.preco}</p>
                    </div>
                    <div className={styles.btnFooter}>
                      <Link to={`/game/${game.id}`}>
                        <Button>Add to cart</Button>
                      </Link>
                      <Link to={`/game/${game.id}`}>
                        <Button>Detalhes</Button>
                      </Link>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          ))}
      </section>
    </div>
  );
}

export default BestSellers;
