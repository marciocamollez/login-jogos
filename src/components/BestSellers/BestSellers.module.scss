@import "../../styles/Variables.scss";

.maisVendidos{
    display: flex;
    padding: 0 20px 20px 20px;
    box-sizing: border-box;

    .container{
        padding-left: 20px;
        padding-right: 20px;

        &:first-child{
            padding-left: 0;
        }
        &:last-child{
            padding-right: 0;
        }

        @media(max-width: 930px){
            padding-left: 0px;
            padding-right: 0px;
            padding-bottom: 20px;
        }
    }

    .boxImage{
        position: relative;

        img{
            max-width: 100%;
            display: block;
            border-radius: 6px;
        }

        

        &:hover{ 
            .boxOverlay{
                opacity: 0.9;
                border: 1px solid $green;
            }
        }

        .boxOverlay{
            position: absolute;
            top: 0;
            left: 0;
            right: 0;
            bottom: 0;
            height: 100%;
            width: 100%;
            opacity: 0;
            background-color: #000;
            border-radius: 6px;

            .boxHide{
                display: flex;
                flex-direction: column;
                justify-content: space-between;
                height: 100%;
                padding: 20px;
                box-sizing: border-box;

                p{
                    color: #FFF;
                }
                
                .preco{
                    color: $green;
                }

                .btnFooter{
                    display: flex;
                    justify-content: flex-end;

                    button{
                        background-color: $green;
                        color: #000;
                        text-transform: uppercase;
                        font-size: 0.8rem;
                        margin-left: 10px;

                        &:hover{
                            background-color: $greenHover;
                        }
                    }
                }
            }
        }
    }

    @media(max-width: 930px){
        flex-direction: column;
        align-items: center;
    }
}