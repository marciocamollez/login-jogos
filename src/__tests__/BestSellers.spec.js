import { render, screen } from "@testing-library/react";
import BestSellers from "../components/BestSellers/BestSellers";
import { MemoryRouter } from "react-router-dom";

describe("BestSellers Component", () => {
  test("should render best sellers div", () => {
    render(<BestSellers />, { wrapper: MemoryRouter });
    const el = screen.getByTestId("test-bestsellers");
    expect(el).toBeInTheDocument();
    expect(el).toMatchSnapshot();
    expect(el).toBeTruthy();
  });
});
