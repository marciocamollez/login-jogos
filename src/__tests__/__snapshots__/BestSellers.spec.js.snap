// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BestSellers Component should render best sellers div 1`] = `
<section
  class="maisVendidos"
  data-testid="test-bestsellers"
/>
`;
